app/helpers/heatmap_helper.rb

```diff
@@ -67,92 +67,22 @@ def self.sample_taxons_dict(params, samples, background_id)
     )
 
-    details = top_taxons_details(
-      results_by_pr,
-      num_results,
-      sort_by,
-      species_selected,
-      threshold_filters
-    )
-
-    taxon_ids = details.pluck('tax_id')
-    taxon_ids -= removed_taxon_ids
-
-    unless taxon_ids.empty?
-      # Refetch at genus level using species level
-      parent_ids = species_selected ? [] : HeatmapHelper.fetch_parent_ids(taxon_ids, samples)
-      results_by_pr = HeatmapHelper.fetch_samples_taxons_counts(samples, taxon_ids, parent_ids, background_id)
-    end
-
     HeatmapHelper.samples_taxons_details(
       results_by_pr,
       samples,
-      taxon_ids,
       species_selected,
-      threshold_filters
+      threshold_filters,
+      sort_by,
+      num_results
     )
   end
 
-  def self.top_taxons_details(
-    results_by_pr,
-    num_results,
-    sort_by,
-    species_selected,
-    threshold_filters
-  )
-    sort = ReportHelper.decode_sort_by(sort_by)
-    count_type = sort[:count_type]
-    metric = sort[:metric]
-    candidate_taxons = {}
-    results_by_pr.each do |_pr_id, res|
-      pr = res["pr"]
-      taxon_counts = res["taxon_counts"]
-      sample_id = pr.sample_id
-
-      tax_2d = ReportHelper.taxon_counts_cleanup(taxon_counts)
-      HeatmapHelper.only_species_or_genus_counts!(tax_2d, species_selected)
-
-      rows = []
-      tax_2d.each do |_tax_id, tax_info|
-        rows << tax_info
-      end
-
-      # NOTE: This block of code can probably be all removed because the same
-      # filtering now happens earlier in SQL.
-      HeatmapHelper.compute_aggregate_scores_v2!(rows)
-      rows = rows.select do |row|
-        row["NT"]["maxzscore"] >= MINIMUM_ZSCORE_THRESHOLD &&
-          # Note: these are applied *after* SQL filters, so results may not be
-          # 100% as expected .
-          HeatmapHelper.apply_custom_filters(row, threshold_filters)
-      end
-
-      # Get the top N for each sample. This re-sorts on the same metric as in
-      # fetch_top_taxons SQL. We sort there first for performance.
-      rows.sort_by! { |tax_info| ((tax_info[count_type] || {})[metric] || 0.0) * -1.0 }
-      count = 1
-      rows.each do |row|
-        taxon = if candidate_taxons[row["tax_id"]]
-                  candidate_taxons[row["tax_id"]]
-                else
-                  { "tax_id" => row["tax_id"], "samples" => {} }
-                end
-        taxon["max_aggregate_score"] = row[sort[:count_type]][sort[:metric]] if
-          taxon["max_aggregate_score"].to_f < row[sort[:count_type]][sort[:metric]].to_f
-        taxon["samples"][sample_id] = [count, row["tax_level"], row["NT"]["zscore"], row["NR"]["zscore"]]
-        candidate_taxons[row["tax_id"]] = taxon
-        break if count >= num_results
-        count += 1
-      end
-    end
-
-    candidate_taxons.values.sort_by { |taxon| -1.0 * taxon["max_aggregate_score"].to_f }
-  end
-
-  def self.fetch_top_taxons(
+  def self.fetch_top_taxons( # rubocop:disable Metrics/ParameterLists
```

Comment:
Can you add instead configure this rule with a higher threshold, say 10 ?

Reply:
I actually think this is a good rule as-is, but it is outside the scope of this PR to refactor to fewer params, hence the exception. What do you think?

Reply:
I see it is already 10. So keep it.